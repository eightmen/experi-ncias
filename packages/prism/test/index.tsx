
/**
 * @jest-environment jsdom
 */

import React from 'react'
import renderer from 'react-test-renderer'
import Prism from '../src'

const CODE = `
  console.log('hello, world!')
`

const render = (el: typeof renderer.create.arguments[0]) =>
  renderer.create(el).toJSON()

test('renders a code block', () => {
  const result = render(<Prism className="language-js">{CODE}</Prism>)

  expect(result).toMatchSnapshot()
})

test('renders with no className', () => {
  // @ts-expect-error className is required by the type definition
  const json = render(<Prism children="<h1>Hello</h1>" />)
  expect(json).toMatchSnapshot()
})

test('renders with other languages', () => {
  const json = render(
    <Prism className="language-php" children="<h1>Hello</h1>" />
  )
  expect(json).toMatchSnapshot()
})

const HIGHLIGHT_CODE = `
 console.log('hello, world!') // highlight-line
`

test('highlights inline comment', () => {
  const result = render(<Prism className="language-js">{HIGHLIGHT_CODE}</Prism>)
  expect(result).toMatchSnapshot()
})

const HIGHLIGHT_START_END = `
  // highlight-start
 console.log('hello, world!') 
 // highlight-end
 let other = "no highlight"
`

test('highlight start and end', () => {
  const result = render(
    <Prism className="language-js">{HIGHLIGHT_START_END}</Prism>
  )

  expect(result).toMatchSnapshot()
})

const NO_HIGHLIGHT = `
  // highlight-start
 console.log('hello, world!') 
`

test('no highlight', () => {
  const result = render(<Prism className="language-js">{NO_HIGHLIGHT}</Prism>)

  expect(result).toMatchSnapshot()
})

const MULTIPLE_HIGHLIGHTS_START_END = `
  // highlight-start
  // i am highlighted!
  // highlight-end

  // no highlight here ...

  // highlight-start 
  // i am highlighted, and
  // so am i!
  // highlight-end

  // no highlight here either ...
`

test('multiple highlights start and end', () => {
  const result = render(
    <Prism className="language-js">{MULTIPLE_HIGHLIGHTS_START_END}</Prism>
  )

  expect(result).toMatchSnapshot()
})