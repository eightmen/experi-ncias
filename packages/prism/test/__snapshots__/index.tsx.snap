
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight start and end 1`] = `
<pre
  className="language-js prism-code language-javascript emotion-0"
  style={{}}
>
  <div
    className="token-line highlight"
  >
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token console class-name"
    >
      console
    </span>
    <span
      className="token punctuation"
    >
      .
    </span>
    <span
      className="token method function property-access"
    >
      log
    </span>
    <span
      className="token punctuation"
    >
      (
    </span>
    <span
      className="token string"
    >
      'hello, world!'
    </span>
    <span
      className="token punctuation"
    >
      )
    </span>
    <span
      className="token plain"
    >
       
    </span>
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token keyword"
    >
      let
    </span>
    <span
      className="token plain"
    >
       other 
    </span>
    <span
      className="token operator"
    >
      =
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token string"
    >
      "no highlight"
    </span>
  </div>
</pre>
`;

exports[`highlights inline comment 1`] = `
<pre
  className="language-js prism-code language-javascript emotion-0"
  style={{}}
>
  <div
    className="token-line highlight"
  >
    <span
      className="token console class-name"
    >
      console
    </span>
    <span
      className="token punctuation"
    >
      .
    </span>
    <span
      className="token method function property-access"
    >
      log
    </span>
    <span
      className="token punctuation"
    >
      (
    </span>
    <span
      className="token string"
    >
      'hello, world!'
    </span>
    <span
      className="token punctuation"
    >
      )
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token comment"
    />
  </div>
</pre>
`;

exports[`multiple highlights start and end 1`] = `
.emotion-1 {
  display: inline-block;
}

<pre
  className="language-js prism-code language-javascript emotion-0"
  style={{}}
>
  <div
    className="token-line highlight"
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
    >
      // i am highlighted!
    </span>
    <span
      className="token plain"
    />
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain emotion-1"
    >
      

    </span>
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
    >
      // no highlight here ...
    </span>
    <span
      className="token plain"
    />
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain emotion-1"
    >
      

    </span>
  </div>
  <div
    className="token-line highlight"
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
    >
      // i am highlighted, and
    </span>
    <span
      className="token plain"
    />
  </div>
  <div
    className="token-line highlight"
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
    >
      // so am i!
    </span>
    <span
      className="token plain"
    />
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain emotion-1"
    >
      

    </span>
  </div>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
    >
      // no highlight here either ...
    </span>
  </div>
</pre>
`;

exports[`no highlight 1`] = `
<pre
  className="language-js prism-code language-javascript emotion-0"
  style={{}}
>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token console class-name"
    >
      console
    </span>
    <span
      className="token punctuation"
    >
      .
    </span>
    <span
      className="token method function property-access"
    >
      log
    </span>
    <span
      className="token punctuation"
    >
      (
    </span>
    <span
      className="token string"
    >
      'hello, world!'
    </span>
    <span
      className="token punctuation"
    >
      )
    </span>
  </div>
</pre>
`;

exports[`renders a code block 1`] = `
<pre
  className="language-js prism-code language-javascript emotion-0"
  style={{}}
>
  <div
    className="token-line"
  >
    <span
      className="token console class-name"
    >
      console
    </span>
    <span
      className="token punctuation"
    >
      .
    </span>
    <span
      className="token method function property-access"
    >
      log
    </span>
    <span
      className="token punctuation"
    >
      (
    </span>
    <span
      className="token string"
    >
      'hello, world!'
    </span>
    <span
      className="token punctuation"
    >
      )
    </span>
  </div>
</pre>
`;

exports[`renders with no className 1`] = `
<pre
  className=" prism-code language- emotion-0"
  style={{}}
>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
      &lt;h1&gt;Hello&lt;/h1&gt;
    </span>
  </div>
</pre>
`;

exports[`renders with other languages 1`] = `
<pre
  className="language-php prism-code language-php emotion-0"
  style={{}}
>
  <div
    className="token-line"
  >
    <span
      className="token plain"
    >
      &lt;h1&gt;Hello&lt;/h1&gt;
    </span>
  </div>
</pre>
`;